sdk/keyvault/keyvault-admin/src/challengeAuthenticationCallbacks.ts

```diff
@@ -106,12 +107,20 @@ export function createChallengeCallbacks(): ChallengeCallbacks {
     if (!challenge) {
       throw new Error("Missing challenge.");
     }
-    const parsedChallenge: ParsedWWWAuthenticate = parseWWWAuthenticate(challenge) || [];
+    const parsedChallenge: ParsedWWWAuthenticate = parseWWWAuthenticate(challenge) || {};
 
-    const accessToken = await options.getAccessToken(
-      parsedChallenge.scope ? [parsedChallenge.scope] : scopes,
-      { ...getTokenOptions, tenantId: parsedChallenge.tenantId }
-    );
+    const scope = parsedChallenge.resource
+      ? parsedChallenge.resource + "/.default"
+      : parsedChallenge.scope;
```

Comment:
interesting, I would think `parsedChallenge.scope` if there is one is preferred over `parsedChallenge.resource + "/.default"`?

Reply:
Following .NET's logic here: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs#L91-L99